js/modules/types/attachment.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-env browser */
+
+const MIME = require('./mime');
+const { arrayBufferToBlob, blobToArrayBuffer, dataURLToBlob } = require('blob-util');
+const { autoOrientImage } = require('../auto_orient_image');
+
+// Increment this everytime we change how attachments are processed. This allows us to
+// retroactively update existing attachments. As we add more processing steps, we could
+// design a pipeline that does this incrementally, e.g. from version 0 (unknown) -> 1,
+// 1 --> 2, etc., similar to how we do database migrations:
+const CURRENT_PROCESS_VERSION = 1;
+
+// // Incoming message attachment fields
+// {
+//   id: string
+//   contentType: MIMEType
+//   data: ArrayBuffer
+//   digest: ArrayBuffer
+//   fileName: string
+//   flags: null
+//   key: ArrayBuffer
+//   size: integer
+//   thumbnail: ArrayBuffer
+// }
+
+// // Outgoing message attachment fields
+// {
+//   contentType: MIMEType
+//   data: ArrayBuffer
+//   fileName: string
+//   size: integer
+// }
```

Comment:
This was an interesting observation: Sending and receiving messages resulted in very different amounts of attachment metadata stored in our IndexedDB. I talked to Matthew and he confirmed that in a nutshell, iOS only preserves protobuf data on an opt-in basis, and unused fields are discarded. What are your thoughts on that?

Just documented these for future specification of our data model 😄